pyomo/contrib/gdpopt/util.py

```diff
@@ -144,24 +142,37 @@ def a_logger(str_or_logger):
         return logging.getLogger(str_or_logger)
 
 
-def copy_var_list_values(from_list, to_list, config, skip_stale=False):
-    """Copy variable values from one list to another."""
+def copy_var_list_values(from_list, to_list, config,
+                         skip_stale=False, skip_fixed=True,
+                         ignore_integrality=False):
+    """Copy variable values from one list to another.
+
+    Rounds to Binary/Integer if neccessary
+    Sets to zero for NonNegativeReals if neccessary
+    """
     for v_from, v_to in zip(from_list, to_list):
         if skip_stale and v_from.stale:
             continue  # Skip stale variable values.
+        if skip_fixed and v_to.is_fixed():
+            continue  # Skip fixed variables.
         try:
             v_to.set_value(value(v_from, exception=False))
             if skip_stale:
                 v_to.stale = False
         except ValueError as err:
             err_msg = getattr(err, 'message', str(err))
             var_val = value(v_from)
-            rounded_val = int(round(var_val))
+            rounded_val = round(var_val) if six.PY3 else int(round(var_val))
```

Comment:
is there any harm in having int(round(var_val)) in Python 3, or do we need to have the check?

Reply:
There is no harm in it. When I looked at it I was surprised to see `int(round(var_val))`, because I expected `round(x)` to be always an integer. After looking up and understanding the problem (python2 returns a float there) I decided on this implementation because (at least to me) it points out why we need the `int` cast.

If you would like me to change it back that's okay though.

Reply:
Yeah, may as well just leave it as `int(round(var_val))`. You can pop in a comment if you want to. Leaves the `six` occurrences for when there are larger compatibility concerns.

Reply:
Fixed in 4b8d692595588f00b0e58c82144a81ff4fab36cd